### ISSUE TITLE
installing opengl with newer Windows SDK

### ISSUE THREAD

**Issue by GeertVL** [Microsoft/vcpkg#99]
When I try to do the following:

```
vcpkg install opengl
```

I get the following error:

```
-- CURRENT_INSTALLED_DIR=C:/works/vcpkg/installed/x86-windows
-- DOWNLOADS=C:/works/vcpkg/downloads
-- CURRENT_PACKAGES_DIR=C:/works/vcpkg/packages/opengl_x86-windows
-- CURRENT_BUILDTREES_DIR=C:/works/vcpkg/buildtrees/opengl
-- CURRENT_PORT_DIR=C:/works/vcpkg/ports/opengl/.
CMake Error at ports/opengl/portfile.cmake:6 (message):
  Cannot find Windows 10.0.10240.0 SDK.  File does not exist: C:\Program
  Files (x86)\Windows Kits\10\Lib\10.0.10240.0\um\x86\OpenGL32.Lib
Call Stack (most recent call first):
  scripts/ports.cmake:96 (include)
```

The SDK for that build number of Windows 10 is not installed. But for 10.0.14393.0 and 10.0.10586.0 it is installed.

We could fix it by changing ofcourse the version number in CMake but that does not help other users that did not install the later versions of the Windows SDK.

There is a function **get_os_version_string()** that detects the build number in **metrics.cpp** that we could use in the **build_internal()** function in **commands_installation.cpp**. So if we call the **get_os_version_string()** there. Extract the buildnumber and give it with the command formatting as an extra parameter -DBUILD_NUMBER. We could change the cmake and so it detects the proper SDK. 

Is that ok that I fix it and make a pull request for that? Or do you see an other solution?

**Comment by ras0219-msft** [comment]
I think we should probably just pull the OpenGL library from your newest installed sdk instead of trying to match your OS version specifically.

Related to #8.

**Comment by adam4813** [comment]
Is it even necessary to have opengl as a port? The Wndows SDK include and library path is automatically (as far as I am aware) added to the Visual Studio directories search paths when you install it. Therefore the OpenGL headers and libraries are on the search path already.

To be more illustrative; I am developing a game and use GLFW3 which was installed via vcpkg. I have NOT installed opengl via vcpkg yet I can compile without issue.

**Comment by GeertVL** [comment]
You have got a point there. As it is in the list and did not work I added an issue.

**Comment by adam4813** [comment]
So perhaps the resolution should in fact be to remove the opengl port?

**Comment by ras0219-msft** [comment]
The `opengl` port enables automatic linkage of `opengl32.lib` when using MSBuild, similar to `kernel32.lib`, `user32.lib`, etc. This makes it possible to run something like:
```
vcpkg install sdl2 opengl
```
and be able to copy-paste `SDL2` samples from online into a fresh project without modfiying the link line (they very often implicitly assume `opengl32.lib` will be linked).

For existing projects that already explicitly link `opengl32.lib`, you're absolutely right that it doesn't make any difference.

**Comment by adam4813** [comment]
I am truly unable to find an example of the situation you are describing where they assume it is linked. Can you provide an example link and I will see if I can find a solution for this issue?

**Comment by mattiascibien** [comment]
I think that this could also apply to me. Isn't there a way to get the directory from the environment?

**Comment by ras0219-msft** [comment]
@adam4813 For example, the third result on Google for me is https://gist.github.com/underscorediscovery/46e4f5b4e3e6de7ad50d which is just a cpp file that assumes the opengl functions are available.

@mattiascibien Yes, the Windows SDK will automatically add it to the link path so a simple `opengl32.lib` link reference will work fine. `find_library()` will also work from CMake just fine, with and without the `opengl` package installed.

The only functional difference is to make `opengl32.lib` implicitly linked when using MSBuild (see #306).

**Comment by alexkaratarakis** [comment]
Resolved by: 3ae8349d1be989c82ec7577537429de5a2993c0c
